sdk/core/azure-core/test/perf/inc/azure/core/test/json_test.hpp

```diff
@@ -0,0 +1,328 @@
+      String = j["String"].get<std::string>();
+
+      JsonOptional::SetIfExists<bool>(NullableBoolean, j, "NullableBoolean");
+      JsonOptional::SetIfExists<int8_t>(NullableInt8, j, "NullableInt8");
+      JsonOptional::SetIfExists<int16_t>(NullableInt16, j, "NullableInt16");
+      JsonOptional::SetIfExists<int32_t>(NullableInt32, j, "NullableInt32");
+      JsonOptional::SetIfExists<int64_t>(NullableInt64, j, "NullableInt64");
+      JsonOptional::SetIfExists<uint8_t>(NullableUint8, j, "NullableUint8");
+      JsonOptional::SetIfExists<uint16_t>(NullableUint16, j, "NullableUint16");
+      JsonOptional::SetIfExists<uint32_t>(NullableUint32, j, "NullableUint32");
+      JsonOptional::SetIfExists<uint64_t>(NullableUint64, j, "NullableUint64");
+      JsonOptional::SetIfExists<float>(NullableFloat, j, "NullableFloat");
+      JsonOptional::SetIfExists<double>(NullableDouble, j, "NullableDouble");
+      JsonOptional::SetIfExists<std::string>(NullableString, j, "NullableString");
+
+      Booleans = j["Booleans"].get<std::vector<bool>>();
+      Int8s = j["Int8s"].get<std::vector<int8_t>>();
+      Int16s = j["Int16s"].get<std::vector<int16_t>>();
+      Int32s = j["Int32s"].get<std::vector<int32_t>>();
+      Int64s = j["Int64s"].get<std::vector<int64_t>>();
+      Uint8s = j["Uint8s"].get<std::vector<uint8_t>>();
+      Uint16s = j["Uint16s"].get<std::vector<uint16_t>>();
+      Uint32s = j["Uint32s"].get<std::vector<uint32_t>>();
+      Uint64s = j["Uint64s"].get<std::vector<uint64_t>>();
+      Floats = j["Floats"].get<std::vector<float>>();
+      Doubles = j["Doubles"].get<std::vector<double>>();
+      Strings = j["Strings"].get<std::vector<std::string>>();
+
+      for (auto const& pair : j["Map"].items())
+      {
+        Map[pair.key()] = pair.value().get<std::string>();
+      }
+    }
+
+    JsonTestObject() = default;
+    JsonTestObject(size_t const& vectorSize)
+    {
+      Boolean = true;
+      Int8 = static_cast<int8_t>(1);
+      Int16 = static_cast<int16_t>(2);
+      Int32 = static_cast<int32_t>(3);
+      Int64 = static_cast<int64_t>(4);
+      Uint8 = static_cast<uint8_t>(5);
+      Uint16 = static_cast<uint16_t>(6);
+      Uint32 = static_cast<uint32_t>(7);
+      Uint64 = static_cast<uint64_t>(8);
+      Float = static_cast<float>(9.0);
+      Double = static_cast<double>(10.0);
+      String = "string";
+
+      NullableBoolean = true;
+      NullableInt8 = static_cast<int8_t>(1);
+      NullableInt16 = static_cast<int16_t>(2);
+      NullableInt32 = static_cast<int32_t>(3);
+      NullableInt64 = static_cast<int64_t>(4);
+      NullableUint8 = static_cast<uint8_t>(5);
+      NullableUint16 = static_cast<uint16_t>(6);
+      NullableUint32 = static_cast<uint32_t>(7);
+      NullableUint64 = static_cast<uint64_t>(8);
+      NullableFloat = static_cast<float>(9.0);
+      NullableDouble = static_cast<double>(10.0);
+      NullableString = "string";
+
+      Booleans = std::vector<bool>(vectorSize, true);
```

Comment:
Does it make sense to break this down to a finer granularity?

Reply:
i'm not sure there is a lot of benefit breaking it down further, once i start profiling, if it makes sense to i will break it down, at the end of the day this code is not set in stone .